client/lib/protect-form/index.tsx

```diff
@@ -79,10 +79,14 @@ export interface ProtectedFormProps {
 /*
  * HOC that passes markChanged/markSaved props to the wrapped component instance
  */
-export const protectForm = createHigherOrderComponent< ProtectedFormProps, any >( ( Component ) => {
+export const protectForm = createHigherOrderComponent< ProtectedFormProps >( ( Component ) => {
 	return ( props ) => {
 		const { markChanged, markSaved } = useProtectForm();
 
+		// There is an issue with a type rework in the upstream package which causes
+		// this correct use to fail without an ignore. See https://github.com/WordPress/gutenberg/pull/37795
+		// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+		// @ts-ignore-next-line
```

Comment:
@dmsnell I started getting a type error after updating the wordpress packages, I believe due to this change: https://github.com/WordPress/gutenberg/pull/37795. Is there a better way to handle this? I see that you used ts-ignore in that PR to handle this scenario, but I don't love having to do it 😅

Reply:
Following up from a Slack convo. In theory, this issue will be fixed in an upcoming TS release (plus perhaps an upstream fix?) So for now, a better solution is to use `@ts-expect-error` instead of `@ts-ignore`.